src/messages/message_accumulator.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::{
+    consensus::{AccumulationError, SignatureAccumulator},
+    messages::{AccumulatingMessage, Message, MessageWithBytes, PlainMessage, SrcAuthority},
+    section::SectionProofChain,
+};
+use serde::{Serialize, Serializer};
+
+#[derive(Default)]
+pub struct MessageAccumulator(SignatureAccumulator<Payload>);
+
+impl MessageAccumulator {
+    pub fn add(&mut self, accumulating_msg: AccumulatingMessage) -> Option<MessageWithBytes> {
+        let payload = Payload {
+            content: accumulating_msg.content,
+            proof_chain: accumulating_msg.proof_chain,
+        };
+
+        match self.0.add(payload, accumulating_msg.proof_share) {
+            Ok((payload, proof)) => {
+                // TODO: should we verify that `proof.public_key` is the same as
+                // `proof_chain.last_key()`?
+
+                let msg = Message {
+                    src: SrcAuthority::Section {
+                        prefix: payload.content.src,
+                        signature: proof.signature,
+                        proof: payload.proof_chain,
```

Comment:
so the `proof` needs to be `proof_chain` actually? not something like: the last proof within the proof_chain?
better make the naming consistent to avoid confusion?